src/libraries/System.Runtime.InteropServices/tests/System.Runtime.InteropServices.UnitTests/System/Runtime/InteropServices/Marshal/GenerateGuidForTypeTests.cs

```diff
@@ -41,7 +41,7 @@ public static IEnumerable<object[]> GenerateGuidForType_Valid_TestData()
             yield return new object[] { collectibleType };
         }
 
-        [Theory]
+        [ConditionalTheory(typeof(PlatformDetection), nameof(PlatformDetection.IsReflectionEmitSupported))]
```

Comment:
Could you change GenerateGuidForType_Valid_TestData to not do the last Ref.Emit part if Emit is not available instead? We would want to test it otherwise. NativeAOT has logic to compute a GUID because people sometimes use it outside COM.